pyrsistent/_checked_types.py

```diff
@@ -344,8 +344,12 @@ def _check(self, it):
             self._invariant_errors.extend(error_data)
 
         def add(self, element):
-            self._check([element])
-            self._pmap_evolver[element] = True
+            return self.update([element])
+
+        def update(self, elements):
+            elements = frozenset(elements)
+            self._check(elements)
+            self._pmap_evolver.update(dict(zip(elements, repeat(True))))
```

Comment:
I switched to `fromkeys`, but I'm not going to try and document the strategy right now....mostly because it's copied from the original `set` method and I don't really understand it.